data/pools/transactionPool.go

```diff
@@ -238,6 +242,7 @@ func (pool *TransactionPool) rememberCommit(flush bool) {
 
 	if flush {
 		pool.pendingTxGroups = pool.rememberedTxGroups
+		pool.stateproofOverflowed = false
```

Comment:
for completeness, you could add this to TransactionPool.Reset() too but that is only called when first restoring from a catchpoint

Reply:
Since `Reset` also initializes an empty `pendingTxids`, there will be a flush way before the pool gets filled.
Hence, `stateproofOverflowed` will be set to `false` before the pool gets filled and before it is needed.